backend/src/v2/driver/driver.go

```diff
@@ -400,11 +492,25 @@ func makePodSpecPatch(
 			Env:       userEnvVar,
 		}},
 	}
-	podSpecPatchBytes, err := json.Marshal(podSpec)
+	return podSpec, nil
+}
+
+// Extends the PodSpec to include Kubernetes specific config.
+func extendPodSpecPatch(
+	podSpec *k8score.PodSpec,
+	kubernetesConfig *kubernetesplatform.KubernetesExecutorConfig,
+	dag *metadata.DAG,
+	dagTasks map[string]*metadata.Execution,
+) error {
+	// Get volume mount information
+	volumeMounts, volumes, err := makeVolumeMountPatch(kubernetesConfig.GetPvcMount(), dag, dagTasks)
 	if err != nil {
-		return "", fmt.Errorf("JSON marshaling pod spec patch: %w", err)
+		return fmt.Errorf("failed to extract volume mount info: %w", err)
 	}
-	return string(podSpecPatchBytes), nil
+	// TODO(gkcalat): add nodeSelector once it is added to platform-specific features in PipelineSpec
+	podSpec.Volumes = volumes
+	podSpec.Containers[0].VolumeMounts = volumeMounts
```

Comment:
I wanted to confirm with you that we intentionally hard-coding the volumes to be mounted only to the first container. It might be okay to assume that we always have a single user container and that it goes first for every pod.

Reply:
Yes, I think we can assume one user container for now.